### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by bramvdh91** [bramvdh91/modelica-annex60#74]
We want to test all basic components, hence a call to provide unit tests here.
The total pipe model is already tested in the validation examples, but maybe it is a good idea to add some basic additional tests?
Here's a list of components that would be good to test (we could base this on the existing examples in UseCases):

- [ ] PipeAdiabaticPlugFlow for standstill and reversing flow
- [ ] PipeCore for standstill and reversing flow
- [ ] TimeDelay component
- [ ] HeatLossPipeDelay (some ambiguous naming here too, maybe drop *delay*?)

**Comment by bramvdh91** [comment]
In addition, the unit tests that can be done now run into a lot of errors in pedantic mode and mostly fail to simulate. Any help here is welcome :)

**Comment by bramvdh91** [comment]
see branch ``pipe_issue74_unitTests``

**Comment by bramvdh91** [comment]
Also cleaning up examples in this branch

**Comment by bramvdh91** [comment]
@marcusfuchs would you be able to help here? I'm trying to run unit tests on the Validation package, but I keep getting errors that cause the unit tests to fail (such as non-literal value, unspecified initial conditions,...)

**Comment by marcusfuchs** [comment]
@bramvdh91 You're right, it would be great to get rid of these errors. I hope I find the time to also look into this.

**Comment by bramvdh91** [comment]
@marcusfuchs I was able to remove the "non-literal values" by including ``dp(nominal=2)`` in the instantiation of the ``HydraulicDiameter res`` :) For the initial conditions, some more work is needed, but in any case now the Validation unit tests are running without problem.

**Comment by marcusfuchs** [comment]
Great!
